File: my-folder/problems/insert_delete_getrandom_o(1)/solution.py
class RandomizedSet:

    def __init__(self):
        self.vals = []
        self.idx = {}
        

    def insert(self, val: int) -> bool:
        if val in self.idx:
            return False
        self.idx[val] = len(self.vals)
        self.vals.append(val)
        return True

    def remove(self, val: int) -> bool:
        if val not in self.idx:
            return False
        i = self.idx[val]
        repl = self.vals[-1]
        self.vals[i] = repl
        self.idx[repl] = i
        del self.idx[val]
        self.vals.pop()
        return True

    def getRandom(self) -> int:
        return random.choice(self.vals)


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()

File: my-folder/problems/insert_delete_getrandom_o(1)/solution.cpp
class RandomizedSet {
std::unordered_map<int, size_t> ind;
std::vector<int> vals;
public:
    RandomizedSet() {}
    
    bool insert(int val) {
        if (ind.contains(val)) return false;
        ind[val] = vals.size();
        vals.push_back(val);
        return true;
    }
    
    bool remove(int val) {
        if (!ind.contains(val)) return false;
        size_t i = ind[val];
        ind.erase(val);
        if (i == vals.size()-1) {
            vals.pop_back();
        } else {
            vals[i] = vals[vals.size()-1];
            ind[vals[i]] = i;
            vals.pop_back();
        }
        return true;
    }
    
    int getRandom() {
        size_t i = rand()%vals.size();
        return vals[i];
    }
};

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet* obj = new RandomizedSet();
 * bool param_1 = obj->insert(val);
 * bool param_2 = obj->remove(val);
 * int param_3 = obj->getRandom();
 */